{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Bioinformatics: network biology project"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Crema Fernando, Puy Tapia Alba, Rossini Valerio"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2. Collect basic information about seed genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### we have collected the basic information about seed genes in a csv called ex2, that we have imported as follow:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "df_r = pd.read_csv(\"C:\\\\Users\\\\valer\\\\Documents\\\\datascience\\\\bio_data\\\\ex2.csv\", sep=';')[0:24]\n",
    "df_r.drop([\"Unnamed: 7\", \"Unnamed: 8\", \"Unnamed: 9\", \"Unnamed: 10\",\n",
    "           \"Unnamed: 11\", \"Unnamed: 12\", \"Unnamed: 13\", \"Unnamed: 14\", \"\\ufeff\"], inplace = True, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_r[\"Entrez Gene ID\"] = df_r[\"Entrez Gene ID\"].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>official gene symbol</th>\n",
       "      <th>Uniprot AC</th>\n",
       "      <th>Uniprot ID</th>\n",
       "      <th>protein name</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>description of its function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ADCY3</td>\n",
       "      <td>O60266</td>\n",
       "      <td>ADCY3_HUMAN</td>\n",
       "      <td>Adenylate cyclase type 3</td>\n",
       "      <td>109</td>\n",
       "      <td>Catalyzes the formation of the signaling molec...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CXCR2</td>\n",
       "      <td>P25025</td>\n",
       "      <td>CXCR2_HUMAN</td>\n",
       "      <td>C-X-C chemokine receptor type 2</td>\n",
       "      <td>3579</td>\n",
       "      <td>Receptor for interleukin-8 which is a powerful...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3B</td>\n",
       "      <td>Q9UBC3</td>\n",
       "      <td>DNM3B_HUMAN</td>\n",
       "      <td>DNA (cytosine-5)-methyltransferase 3B</td>\n",
       "      <td>1789</td>\n",
       "      <td>Required for genome-wide de novo methylation a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FAP</td>\n",
       "      <td>Q12884</td>\n",
       "      <td>SEPR_HUMAN</td>\n",
       "      <td>Prolyl endopeptidase FAP</td>\n",
       "      <td>2191</td>\n",
       "      <td>Cell surface glycoprotein serine protease that...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FOS</td>\n",
       "      <td>P01100</td>\n",
       "      <td>FOS_HUMAN</td>\n",
       "      <td>Proto-oncogene c-Fos</td>\n",
       "      <td>2353</td>\n",
       "      <td>Nuclear phosphoprotein which forms a tight but...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>FUT2</td>\n",
       "      <td>\\tQ10981</td>\n",
       "      <td>FUT2_HUMAN</td>\n",
       "      <td>Galactoside 2-alpha-L-fucosyltransferase 2</td>\n",
       "      <td>2524</td>\n",
       "      <td>Mediates the transfer of fucose to the termina...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GPR35</td>\n",
       "      <td>Q9HC97</td>\n",
       "      <td>GPR35_HUMAN</td>\n",
       "      <td>G-protein coupled receptor 35</td>\n",
       "      <td>2859</td>\n",
       "      <td>Acts as a receptor for kynurenic acid, an inte...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>IFIH1</td>\n",
       "      <td>Q9BYX4</td>\n",
       "      <td>IFIH1_HUMAN</td>\n",
       "      <td>Interferon-induced helicase C domain-containin...</td>\n",
       "      <td>64135</td>\n",
       "      <td>Innate immune receptor which acts as a cytopla...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>IL23R</td>\n",
       "      <td>Q5VWK5</td>\n",
       "      <td>IL23R_HUMAN</td>\n",
       "      <td>Interleukin-23 receptor</td>\n",
       "      <td>149233</td>\n",
       "      <td>Associates with IL12RB1 to form the interleuki...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>IL27</td>\n",
       "      <td>Q8NEV9</td>\n",
       "      <td>IL27A_HUMAN</td>\n",
       "      <td>Interleukin-27 subunit alpha</td>\n",
       "      <td>246778</td>\n",
       "      <td>Associates with EBI3 to form the IL-27 interle...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>IL2RA</td>\n",
       "      <td>P01589</td>\n",
       "      <td>IL2RA_HUMAN</td>\n",
       "      <td>Interleukin-2 receptor subunit alpha</td>\n",
       "      <td>3559</td>\n",
       "      <td>Receptor for interleukin-2. The receptor is in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>KEAP1</td>\n",
       "      <td>\\tQ14145</td>\n",
       "      <td>KEAP1_HUMAN</td>\n",
       "      <td>Kelch-like ECH-associated protein 1</td>\n",
       "      <td>9817</td>\n",
       "      <td>Acts as a substrate adapter protein for the E3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>LCE3B</td>\n",
       "      <td>\\tQ5TA77</td>\n",
       "      <td>LCE3B_HUMAN</td>\n",
       "      <td>Late cornified envelope protein 3B</td>\n",
       "      <td>353143</td>\n",
       "      <td>Precursors of the cornified envelope of the st...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NFKB1</td>\n",
       "      <td>\\tP19838</td>\n",
       "      <td>NFKB1_HUMAN</td>\n",
       "      <td>Nuclear factor NF-kappa-B p105 subunit</td>\n",
       "      <td>4790</td>\n",
       "      <td>NF-kappa-B is a pleiotropic transcription fact...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NXPE1</td>\n",
       "      <td>Q8N323</td>\n",
       "      <td>NXPE1_HUMAN</td>\n",
       "      <td>NXPE family member 1</td>\n",
       "      <td>120400</td>\n",
       "      <td>Not Found in uniprot</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>OR5B21</td>\n",
       "      <td>A6NL26</td>\n",
       "      <td>OR5BL_HUMAN</td>\n",
       "      <td>Olfactory receptor 5B21</td>\n",
       "      <td>219968</td>\n",
       "      <td>Odorant receptor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>Q99650</td>\n",
       "      <td>OSMR_HUMAN</td>\n",
       "      <td>Oncostatin-M-specific receptor subunit beta</td>\n",
       "      <td>9180</td>\n",
       "      <td>Associates with IL31RA to form the IL31 recept...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>P17706</td>\n",
       "      <td>PTN2_HUMAN</td>\n",
       "      <td>Tyrosine-protein phosphatase non-receptor type 2</td>\n",
       "      <td>5771</td>\n",
       "      <td>Non-receptor type tyrosine-specific phosphatas...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>RAVER1</td>\n",
       "      <td>Q8IY67</td>\n",
       "      <td>RAVR1_HUMAN</td>\n",
       "      <td>Ribonucleoprotein PTB-binding 1</td>\n",
       "      <td>125950</td>\n",
       "      <td>Cooperates with PTBP1 to modulate regulated al...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>RNF186</td>\n",
       "      <td>Q9NXI6</td>\n",
       "      <td>RN186_HUMAN</td>\n",
       "      <td>RING finger protein 186</td>\n",
       "      <td>54546</td>\n",
       "      <td>Not provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>RPS6KB1</td>\n",
       "      <td>P23443</td>\n",
       "      <td>KS6B1_HUMAN</td>\n",
       "      <td>Ribosomal protein S6 kinase beta-1</td>\n",
       "      <td>6198</td>\n",
       "      <td>Serine/threonine-protein kinase that acts down...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SH2B3</td>\n",
       "      <td>Q9UQQ2</td>\n",
       "      <td>Q9UQQ2_HUMAN</td>\n",
       "      <td>SH2B adapter protein 3</td>\n",
       "      <td>10019</td>\n",
       "      <td>Links T-cell receptor activation signal to pho...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>TNFRSF6B</td>\n",
       "      <td>O95407</td>\n",
       "      <td>O95407_HUMAN</td>\n",
       "      <td>Tumor necrosis factor receptor superfamily mem...</td>\n",
       "      <td>8771</td>\n",
       "      <td>Decoy receptor that can neutralize the cytotox...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TYK2</td>\n",
       "      <td>P29597</td>\n",
       "      <td>P29597_HUMAN</td>\n",
       "      <td>Non-receptor tyrosine-protein kinase TYK2</td>\n",
       "      <td>7297</td>\n",
       "      <td>Probably involved in intracellular signal tran...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   official gene symbol      Uniprot AC   Uniprot ID   \\\n",
       "0                 ADCY3          O60266   ADCY3_HUMAN   \n",
       "1                 CXCR2          P25025   CXCR2_HUMAN   \n",
       "2                DNMT3B          Q9UBC3   DNM3B_HUMAN   \n",
       "3                   FAP          Q12884    SEPR_HUMAN   \n",
       "4                   FOS          P01100     FOS_HUMAN   \n",
       "5                  FUT2        \\tQ10981    FUT2_HUMAN   \n",
       "6                 GPR35          Q9HC97   GPR35_HUMAN   \n",
       "7                IFIH1           Q9BYX4   IFIH1_HUMAN   \n",
       "8                 IL23R          Q5VWK5   IL23R_HUMAN   \n",
       "9                  IL27          Q8NEV9   IL27A_HUMAN   \n",
       "10                IL2RA          P01589   IL2RA_HUMAN   \n",
       "11                KEAP1        \\tQ14145   KEAP1_HUMAN   \n",
       "12                LCE3B        \\tQ5TA77   LCE3B_HUMAN   \n",
       "13                NFKB1        \\tP19838   NFKB1_HUMAN   \n",
       "14                NXPE1          Q8N323   NXPE1_HUMAN   \n",
       "15               OR5B21          A6NL26   OR5BL_HUMAN   \n",
       "16                 OSMR          Q99650    OSMR_HUMAN   \n",
       "17                PTPN2          P17706    PTN2_HUMAN   \n",
       "18               RAVER1          Q8IY67   RAVR1_HUMAN   \n",
       "19               RNF186          Q9NXI6   RN186_HUMAN   \n",
       "20              RPS6KB1          P23443   KS6B1_HUMAN   \n",
       "21                SH2B3          Q9UQQ2  Q9UQQ2_HUMAN   \n",
       "22             TNFRSF6B          O95407  O95407_HUMAN   \n",
       "23                 TYK2          P29597  P29597_HUMAN   \n",
       "\n",
       "                                         protein name  Entrez Gene ID  \\\n",
       "0                            Adenylate cyclase type 3             109   \n",
       "1                     C-X-C chemokine receptor type 2            3579   \n",
       "2               DNA (cytosine-5)-methyltransferase 3B            1789   \n",
       "3                            Prolyl endopeptidase FAP            2191   \n",
       "4                                Proto-oncogene c-Fos            2353   \n",
       "5          Galactoside 2-alpha-L-fucosyltransferase 2            2524   \n",
       "6                       G-protein coupled receptor 35            2859   \n",
       "7   Interferon-induced helicase C domain-containin...           64135   \n",
       "8                             Interleukin-23 receptor          149233   \n",
       "9                        Interleukin-27 subunit alpha          246778   \n",
       "10               Interleukin-2 receptor subunit alpha            3559   \n",
       "11                Kelch-like ECH-associated protein 1            9817   \n",
       "12                 Late cornified envelope protein 3B          353143   \n",
       "13             Nuclear factor NF-kappa-B p105 subunit            4790   \n",
       "14                               NXPE family member 1          120400   \n",
       "15                            Olfactory receptor 5B21          219968   \n",
       "16        Oncostatin-M-specific receptor subunit beta            9180   \n",
       "17   Tyrosine-protein phosphatase non-receptor type 2            5771   \n",
       "18                    Ribonucleoprotein PTB-binding 1          125950   \n",
       "19                            RING finger protein 186           54546   \n",
       "20                 Ribosomal protein S6 kinase beta-1            6198   \n",
       "21                             SH2B adapter protein 3           10019   \n",
       "22  Tumor necrosis factor receptor superfamily mem...            8771   \n",
       "23          Non-receptor tyrosine-protein kinase TYK2            7297   \n",
       "\n",
       "                          description of its function  \n",
       "0   Catalyzes the formation of the signaling molec...  \n",
       "1   Receptor for interleukin-8 which is a powerful...  \n",
       "2   Required for genome-wide de novo methylation a...  \n",
       "3   Cell surface glycoprotein serine protease that...  \n",
       "4   Nuclear phosphoprotein which forms a tight but...  \n",
       "5   Mediates the transfer of fucose to the termina...  \n",
       "6   Acts as a receptor for kynurenic acid, an inte...  \n",
       "7   Innate immune receptor which acts as a cytopla...  \n",
       "8   Associates with IL12RB1 to form the interleuki...  \n",
       "9   Associates with EBI3 to form the IL-27 interle...  \n",
       "10  Receptor for interleukin-2. The receptor is in...  \n",
       "11  Acts as a substrate adapter protein for the E3...  \n",
       "12  Precursors of the cornified envelope of the st...  \n",
       "13  NF-kappa-B is a pleiotropic transcription fact...  \n",
       "14                               Not Found in uniprot  \n",
       "15                                   Odorant receptor  \n",
       "16  Associates with IL31RA to form the IL31 recept...  \n",
       "17  Non-receptor type tyrosine-specific phosphatas...  \n",
       "18  Cooperates with PTBP1 to modulate regulated al...  \n",
       "19                                       Not provided  \n",
       "20  Serine/threonine-protein kinase that acts down...  \n",
       "21  Links T-cell receptor activation signal to pho...  \n",
       "22  Decoy receptor that can neutralize the cytotox...  \n",
       "23  Probably involved in intracellular signal tran...  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_r"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3. Collect interaction data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### For each seed gene, we need to collect all binary protein interactions from three different PPI sources, so we have written some functions to collect them. we also kept them in a dataframe described below"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Retriving the current directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\valer\\Documents\\datascience\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "cwd = os.getcwd()\n",
    "print(cwd)\n",
    "current_folder = \"APID\"\n",
    "print(cwd + \"\\\\bio_data\\\\\" + current_folder)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# General function to retrieve files in dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def read_files_from_dir(path, c_dir, format_file=\".tsv\"):\n",
    "    from os import listdir\n",
    "    from os.path import isfile, join\n",
    "\n",
    "    mypath = path + c_dir\n",
    "\n",
    "    onlyfiles = [f for f in listdir(mypath) if isfile(join(mypath, f))]\n",
    "\n",
    "    files_to_open = list(filter(lambda x: format_file in x, onlyfiles))\n",
    "    \n",
    "    return files_to_open"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# General function to read files and concatenate interaction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def create_interactions(path, c_dir, columns, files, name):\n",
    "    import pandas as pd\n",
    "\n",
    "    frames = []\n",
    "\n",
    "    for i in files:\n",
    "        s = path+c_dir+\"\\\\\"+i\n",
    "        print(s)\n",
    "        frames.append(pd.read_csv(s, sep=\"\\t\", header=0)[columns])\n",
    "\n",
    "    result = pd.concat(frames)\n",
    "\n",
    "    result.reset_index(inplace=True)\n",
    "\n",
    "    result.drop(\"index\", axis=1, inplace=True)\n",
    "\n",
    "    result.to_csv(path+\"\\\\\"+name, sep=\"\\t\", header=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Treating APID folder"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### we dowload from APID web page all the information of the interations of the seed genes provided and merge them in a table called apid_all.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['ADCY3.txt', 'CXCR2.txt', 'DNMT3B.txt', 'FAP.txt', 'FOS.txt', 'FUT2.txt', 'GPR35.txt', 'IFIH1.txt', 'IL23R.txt', 'IL27.txt', 'IL2RA.txt', 'KEAP1.txt', 'LCE3B.txt', 'NFKB1.txt', 'OSMR.txt', 'PTPN2.txt', 'RAVER1.txt', 'RNF186.txt', 'RPS6KB1.txt', 'SH2B3.txt', 'TNFRSF6B.txt', 'TYK2.txt']\n"
     ]
    }
   ],
   "source": [
    "to_open_APID = read_files_from_dir(path=cwd + \"\\\\bio_data\\\\\",c_dir=\"APID\", format_file=\".txt\")\n",
    "                   \n",
    "print(to_open_APID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\ADCY3.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\CXCR2.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\DNMT3B.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\FAP.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\FOS.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\FUT2.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\GPR35.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\IFIH1.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\IL23R.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\IL27.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\IL2RA.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\KEAP1.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\LCE3B.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\NFKB1.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\OSMR.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\PTPN2.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\RAVER1.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\RNF186.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\RPS6KB1.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\SH2B3.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\TNFRSF6B.txt\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\APID\\TYK2.txt\n"
     ]
    }
   ],
   "source": [
    "create_interactions(path=cwd + \"\\\\bio_data\\\\\",\n",
    "                    c_dir=\"APID\", \n",
    "                    columns=[\"UniprotID_A\",\"UniprotID_B\", \"GeneName_A\", \"GeneName_B\"], \n",
    "                    files=to_open_APID, \n",
    "                    name=\"apid_all.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'C:\\\\Users\\\\valer\\\\Documents\\\\datascience\\\\bio_data\\\\'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cwd + \"\\\\bio_data\\\\\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Treating STRING\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### we dowload from STRING web page all the information of the interations of the seed genes provided and merge them in a table called string_all.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\n"
     ]
    }
   ],
   "source": [
    "current_folder = \"STRING\"\n",
    "print(cwd + \"\\\\bio_data\\\\\" + current_folder)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['ADCY3.tsv', 'all.tsv', 'CXCR2.tsv', 'DNMT3B.tsv', 'FAP.tsv', 'FOS.tsv', 'FUT2.tsv', 'GPR35.tsv', 'IFIH1.tsv', 'IL23R.tsv', 'IL27.tsv', 'IL2RA.tsv', 'KEAP1.tsv', 'NFKB1.tsv', 'NXPE1.tsv', 'OR5B21.tsv', 'OSMR.tsv', 'PTPN2.tsv', 'RAVER1.tsv', 'RPS6KB1.tsv', 'SH2B3.tsv', 'TNFRSF6B.tsv', 'TYK2.tsv']\n"
     ]
    }
   ],
   "source": [
    "to_open_STRING = read_files_from_dir(path=cwd + \"\\\\bio_data\\\\\",c_dir=\"STRING\", format_file=\".tsv\")\n",
    "                   \n",
    "print(to_open_STRING)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\ADCY3.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\all.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\CXCR2.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\DNMT3B.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\FAP.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\FOS.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\FUT2.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\GPR35.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\IFIH1.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\IL23R.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\IL27.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\IL2RA.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\KEAP1.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\NFKB1.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\NXPE1.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\OR5B21.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\OSMR.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\PTPN2.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\RAVER1.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\RPS6KB1.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\SH2B3.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\TNFRSF6B.tsv\n",
      "C:\\Users\\valer\\Documents\\datascience\\bio_data\\STRING\\TYK2.tsv\n"
     ]
    }
   ],
   "source": [
    "create_interactions(path=cwd + \"\\\\bio_data\\\\\",\n",
    "                    c_dir=\"STRING\", \n",
    "                    columns=[\"#node1\",\"node2\"], \n",
    "                    files=to_open_STRING, \n",
    "                    name=\"string_all.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "biiiiiiiiiiiiiiiiiooooooooooooooooooooooo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Summarize the main results in a table reporting:\n",
    "    a) no. of seed genes found in each different DBs (some seed genes may be missing\n",
    "    in some of the DBs);\n",
    "    b) total no. of interacting proteins, including seed genes, for each DB;\n",
    "    c) total no. of interactions found in each DB."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### first we are going to count the total number of interating proteins including seed genes for each database"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### APID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniprotID_A</th>\n",
       "      <th>UniprotID_B</th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q13555</td>\n",
       "      <td>O60266</td>\n",
       "      <td>CAMK2G</td>\n",
       "      <td>ADCY3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q8TAV4</td>\n",
       "      <td>O60266</td>\n",
       "      <td>STOML3</td>\n",
       "      <td>ADCY3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P07550</td>\n",
       "      <td>O60266</td>\n",
       "      <td>ADRB2</td>\n",
       "      <td>ADCY3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q9HBU9</td>\n",
       "      <td>O60266</td>\n",
       "      <td>POPDC2</td>\n",
       "      <td>ADCY3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P17931</td>\n",
       "      <td>O60266</td>\n",
       "      <td>LGALS3</td>\n",
       "      <td>ADCY3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  UniprotID_A UniprotID_B GeneName_A GeneName_B\n",
       "0      Q13555      O60266     CAMK2G      ADCY3\n",
       "1      Q8TAV4      O60266     STOML3      ADCY3\n",
       "2      P07550      O60266      ADRB2      ADCY3\n",
       "3      Q9HBU9      O60266     POPDC2      ADCY3\n",
       "4      P17931      O60266     LGALS3      ADCY3"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_APID = pd.read_csv(\"C:\\\\Users\\\\valer\\\\Documents\\\\datascience\\\\bio_data\\\\apid_all.tsv\", sep=\"\\t\")\n",
    "df_APID.drop(\"Unnamed: 0\", inplace = True, axis = 1)\n",
    "df_APID.columns\n",
    "df_APID.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "Autoinflammatory_diseases =[\"ADCY3\", \"CXCR2\", \"DNMT3B\", \"FAP\", \"FOS\", \"FUT2\", \"GPR35\", \"IFIH1\", \n",
    "                            \"IL23R\", \"IL27\", \"IL2RA\", \"KEAP1\", \"LCE3B\", \"NFKB1\", \"NXPE1\", \"OR5B21\",\n",
    "                            \"OSMR\", \"PTPN2\", \"RAVER1\", \"RNF186\", \"RPS6KB1\", \"SH2B3\", \"TNFRSF6B\", \"TYK2\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### these are proteins-proteins interations "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "a=df_APID[(df_APID[\"GeneName_A\"].isin(Autoinflammatory_diseases)==False)] \n",
    "b = a[(a[\"GeneName_B\"].isin(Autoinflammatory_diseases)==False)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### these are proteins-genes interations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "c=df_APID[(df_APID[\"GeneName_A\"].isin(Autoinflammatory_diseases)==False)] \n",
    "d = c[(c[\"GeneName_B\"].isin(Autoinflammatory_diseases)==True)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### these are genes-proteins interations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "e=df_APID[(df_APID[\"GeneName_A\"].isin(Autoinflammatory_diseases)==True)] \n",
    "f = e[(e[\"GeneName_B\"].isin(Autoinflammatory_diseases)==False)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### we merged these three interations in a single dataframe called df_interacting_proteins_APID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_interacting_proteins_APID = pd.concat([b,d,f])\n",
    "df_interacting_proteins_APID.reset_index(inplace=True)\n",
    "df_interacting_proteins_APID.drop(\"index\", inplace = True, axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### STRING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#node1</th>\n",
       "      <th>node2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PRKACB</td>\n",
       "      <td>PRKACA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDE4B</td>\n",
       "      <td>PRKACA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRKACB</td>\n",
       "      <td>ADCY3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRKACA</td>\n",
       "      <td>ADCY3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PRKACG</td>\n",
       "      <td>ADCY3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #node1   node2\n",
       "0  PRKACB  PRKACA\n",
       "1   PDE4B  PRKACA\n",
       "2  PRKACB   ADCY3\n",
       "3  PRKACA   ADCY3\n",
       "4  PRKACG   ADCY3"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_STRING = pd.read_csv(\"C:\\\\Users\\\\valer\\\\Documents\\\\datascience\\\\bio_data\\\\string_all.tsv\", sep=\"\\t\")\n",
    "df_STRING.drop(\"Unnamed: 0\", inplace = True, axis = 1)\n",
    "df_STRING.columns\n",
    "df_STRING.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### these are proteins-proteins interations "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "a=df_STRING[(df_STRING[\"#node1\"].isin(Autoinflammatory_diseases)==False)] \n",
    "b = a[(a[\"node2\"].isin(Autoinflammatory_diseases)==False)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### these are proteins-genes interations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "c=df_STRING[(df_STRING[\"#node1\"].isin(Autoinflammatory_diseases)==False)] \n",
    "d = c[(c[\"node2\"].isin(Autoinflammatory_diseases)==True)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### these are genes-proteins interations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "e=df_STRING[(df_STRING[\"#node1\"].isin(Autoinflammatory_diseases)==True)] \n",
    "f = e[(e[\"node2\"].isin(Autoinflammatory_diseases)==False)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### we merged these three interations in a single dataframe called df_interacting_proteins_STRING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#node1</th>\n",
       "      <th>node2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PRKACB</td>\n",
       "      <td>PRKACA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDE4B</td>\n",
       "      <td>PRKACA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRKACB</td>\n",
       "      <td>PDE4B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRKACG</td>\n",
       "      <td>PDE4B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PRKACG</td>\n",
       "      <td>PRKACA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #node1   node2\n",
       "0  PRKACB  PRKACA\n",
       "1   PDE4B  PRKACA\n",
       "2  PRKACB   PDE4B\n",
       "3  PRKACG   PDE4B\n",
       "4  PRKACG  PRKACA"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_interacting_proteins_STRING = pd.concat([b,d,f])\n",
    "df_interacting_proteins_STRING.reset_index(inplace=True)\n",
    "df_interacting_proteins_STRING.drop(\"index\", inplace = True, axis = 1)\n",
    "df_interacting_proteins_STRING.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "biiiiiiiiiiiiiiiiiiiiiiooooooooooooooooooooooooooooo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### to compute the number of seed genes in each different database, we used add the number of files that we have download from each webpage. to compute the number of interation from each database we just counted the rows at the _all_ table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>3.3.A Number of seed genes</th>\n",
       "      <th>3.3.B Number of interating proteins</th>\n",
       "      <th>3.3.C Number of interations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>APID</th>\n",
       "      <td>22</td>\n",
       "      <td>9779</td>\n",
       "      <td>9802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STRING</th>\n",
       "      <td>23</td>\n",
       "      <td>1440</td>\n",
       "      <td>1446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BIOGRID</th>\n",
       "      <td>100000</td>\n",
       "      <td>100000</td>\n",
       "      <td>100000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         3.3.A Number of seed genes  3.3.B Number of interating proteins  \\\n",
       "APID                             22                                 9779   \n",
       "STRING                           23                                 1440   \n",
       "BIOGRID                      100000                               100000   \n",
       "\n",
       "         3.3.C Number of interations  \n",
       "APID                            9802  \n",
       "STRING                          1446  \n",
       "BIOGRID                       100000  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = {'3.3.A Number of seed genes' : [len(to_open_APID), len(to_open_STRING), 100000], \n",
    "     '3.3.B Number of interating proteins' : [df_interacting_proteins_APID.shape[0], df_interacting_proteins_STRING.shape[0], 100000],\n",
    "    '3.3.C Number of interations' : [df_APID.shape[0], df_STRING.shape[0], 100000]}\n",
    "pd.DataFrame(d, index=['APID', 'STRING', 'BIOGRID'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. arrange interation data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "a=df_APID[(df_APID[\"GeneName_A\"].isin(Autoinflammatory_diseases)==True)] \n",
    "b = a[(a[\"GeneName_B\"].isin(Autoinflammatory_diseases)==True)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "aa=df_STRING[(df_STRING[\"#node1\"].isin(Autoinflammatory_diseases)==True)] \n",
    "bb = aa[(aa[\"node2\"].isin(Autoinflammatory_diseases)==True)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniprotID_A</th>\n",
       "      <th>UniprotID_B</th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>P25025</td>\n",
       "      <td>P25025</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>CXCR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>644</th>\n",
       "      <td>Q12884</td>\n",
       "      <td>Q12884</td>\n",
       "      <td>FAP</td>\n",
       "      <td>FAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>814</th>\n",
       "      <td>P01100</td>\n",
       "      <td>P01100</td>\n",
       "      <td>FOS</td>\n",
       "      <td>FOS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>855</th>\n",
       "      <td>P01100</td>\n",
       "      <td>P19838</td>\n",
       "      <td>FOS</td>\n",
       "      <td>NFKB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1229</th>\n",
       "      <td>P19838</td>\n",
       "      <td>P19838</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>NFKB1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     UniprotID_A UniprotID_B GeneName_A GeneName_B\n",
       "61        P25025      P25025      CXCR2      CXCR2\n",
       "644       Q12884      Q12884        FAP        FAP\n",
       "814       P01100      P01100        FOS        FOS\n",
       "855       P01100      P19838        FOS      NFKB1\n",
       "1229      P19838      P19838      NFKB1      NFKB1"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniprotID_A</th>\n",
       "      <th>UniprotID_B</th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>P25025</td>\n",
       "      <td>P25025</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>644</th>\n",
       "      <td>Q12884</td>\n",
       "      <td>Q12884</td>\n",
       "      <td>FAP</td>\n",
       "      <td>FAP</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>814</th>\n",
       "      <td>P01100</td>\n",
       "      <td>P01100</td>\n",
       "      <td>FOS</td>\n",
       "      <td>FOS</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>855</th>\n",
       "      <td>P01100</td>\n",
       "      <td>P19838</td>\n",
       "      <td>FOS</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1229</th>\n",
       "      <td>P19838</td>\n",
       "      <td>P19838</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     UniprotID_A UniprotID_B GeneName_A GeneName_B database\n",
       "61        P25025      P25025      CXCR2      CXCR2     APID\n",
       "644       Q12884      Q12884        FAP        FAP     APID\n",
       "814       P01100      P01100        FOS        FOS     APID\n",
       "855       P01100      P19838        FOS      NFKB1     APID\n",
       "1229      P19838      P19838      NFKB1      NFKB1     APID"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b[\"database\"] = [\"APID\"]*b.shape[0]\n",
    "b.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#node1</th>\n",
       "      <th>node2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>IL27</td>\n",
       "      <td>TYK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>TYK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>610</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>TYK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>IL27</td>\n",
       "      <td>TYK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1257</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>TYK2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     #node1 node2\n",
       "359    IL27  TYK2\n",
       "559    OSMR  TYK2\n",
       "610   PTPN2  TYK2\n",
       "1057   IL27  TYK2\n",
       "1257   OSMR  TYK2"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#node1</th>\n",
       "      <th>node2</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>IL27</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>610</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>IL27</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1257</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1308</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     #node1 node2 database\n",
       "359    IL27  TYK2   STRING\n",
       "559    OSMR  TYK2   STRING\n",
       "610   PTPN2  TYK2   STRING\n",
       "1057   IL27  TYK2   STRING\n",
       "1257   OSMR  TYK2   STRING\n",
       "1308  PTPN2  TYK2   STRING"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bb[\"database\"] = [\"STRING\"]*bb.shape[0]\n",
    "bb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "cc=bb.copy()\n",
    "cc.rename(columns={'#node1': 'GeneName_A', 'node2': 'GeneName_B'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>IL27</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>610</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>IL27</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1257</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1308</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     GeneName_A GeneName_B database\n",
       "359        IL27       TYK2   STRING\n",
       "559        OSMR       TYK2   STRING\n",
       "610       PTPN2       TYK2   STRING\n",
       "1057       IL27       TYK2   STRING\n",
       "1257       OSMR       TYK2   STRING\n",
       "1308      PTPN2       TYK2   STRING"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Q8NEV9', 'Q99650', 'P17706', 'Q8NEV9', 'Q99650', 'P17706']"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "u_list = []\n",
    "for i in cc[\"GeneName_A\"].values:\n",
    "    u_list.append(df_r[df_r[\"official gene symbol\"]==i][\"Uniprot AC\"].values[0])\n",
    "u_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "      <th>database</th>\n",
       "      <th>UniprotID_A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>IL27</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "      <td>Q8NEV9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "      <td>Q99650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>610</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "      <td>P17706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>IL27</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "      <td>Q8NEV9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1257</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "      <td>Q99650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1308</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "      <td>P17706</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     GeneName_A GeneName_B database UniprotID_A\n",
       "359        IL27       TYK2   STRING      Q8NEV9\n",
       "559        OSMR       TYK2   STRING      Q99650\n",
       "610       PTPN2       TYK2   STRING      P17706\n",
       "1057       IL27       TYK2   STRING      Q8NEV9\n",
       "1257       OSMR       TYK2   STRING      Q99650\n",
       "1308      PTPN2       TYK2   STRING      P17706"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cc[\"UniprotID_A\"] = u_list\n",
    "cc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['P29597', 'P29597', 'P29597', 'P29597', 'P29597', 'P29597']"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "u1_list = []\n",
    "for i in cc[\"GeneName_B\"].values:\n",
    "    u1_list.append(df_r[df_r[\"official gene symbol\"]==i][\"Uniprot AC\"].values[0])\n",
    "u1_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "      <th>database</th>\n",
       "      <th>UniprotID_A</th>\n",
       "      <th>UniprotID_B</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>IL27</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "      <td>Q8NEV9</td>\n",
       "      <td>P29597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "      <td>Q99650</td>\n",
       "      <td>P29597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>610</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "      <td>P17706</td>\n",
       "      <td>P29597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>IL27</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "      <td>Q8NEV9</td>\n",
       "      <td>P29597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1257</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "      <td>Q99650</td>\n",
       "      <td>P29597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1308</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "      <td>P17706</td>\n",
       "      <td>P29597</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     GeneName_A GeneName_B database UniprotID_A UniprotID_B\n",
       "359        IL27       TYK2   STRING      Q8NEV9      P29597\n",
       "559        OSMR       TYK2   STRING      Q99650      P29597\n",
       "610       PTPN2       TYK2   STRING      P17706      P29597\n",
       "1057       IL27       TYK2   STRING      Q8NEV9      P29597\n",
       "1257       OSMR       TYK2   STRING      Q99650      P29597\n",
       "1308      PTPN2       TYK2   STRING      P17706      P29597"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cc[\"UniprotID_B\"] = u1_list\n",
    "cc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['GeneName_A', 'GeneName_B', 'database', 'UniprotID_A', 'UniprotID_B'], dtype='object')"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cc.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "columns_to_change = [\"UniprotID_A\", \"UniprotID_B\", \"GeneName_A\", \"GeneName_B\",\"database\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "cc = cc[columns_to_change]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniprotID_A</th>\n",
       "      <th>UniprotID_B</th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>Q8NEV9</td>\n",
       "      <td>P29597</td>\n",
       "      <td>IL27</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>Q99650</td>\n",
       "      <td>P29597</td>\n",
       "      <td>OSMR</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>610</th>\n",
       "      <td>P17706</td>\n",
       "      <td>P29597</td>\n",
       "      <td>PTPN2</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>Q8NEV9</td>\n",
       "      <td>P29597</td>\n",
       "      <td>IL27</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1257</th>\n",
       "      <td>Q99650</td>\n",
       "      <td>P29597</td>\n",
       "      <td>OSMR</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1308</th>\n",
       "      <td>P17706</td>\n",
       "      <td>P29597</td>\n",
       "      <td>PTPN2</td>\n",
       "      <td>TYK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     UniprotID_A UniprotID_B GeneName_A GeneName_B database\n",
       "359       Q8NEV9      P29597       IL27       TYK2   STRING\n",
       "559       Q99650      P29597       OSMR       TYK2   STRING\n",
       "610       P17706      P29597      PTPN2       TYK2   STRING\n",
       "1057      Q8NEV9      P29597       IL27       TYK2   STRING\n",
       "1257      Q99650      P29597       OSMR       TYK2   STRING\n",
       "1308      P17706      P29597      PTPN2       TYK2   STRING"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniprotID_A</th>\n",
       "      <th>UniprotID_B</th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>P25025</td>\n",
       "      <td>P25025</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>644</th>\n",
       "      <td>Q12884</td>\n",
       "      <td>Q12884</td>\n",
       "      <td>FAP</td>\n",
       "      <td>FAP</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>814</th>\n",
       "      <td>P01100</td>\n",
       "      <td>P01100</td>\n",
       "      <td>FOS</td>\n",
       "      <td>FOS</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>855</th>\n",
       "      <td>P01100</td>\n",
       "      <td>P19838</td>\n",
       "      <td>FOS</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1229</th>\n",
       "      <td>P19838</td>\n",
       "      <td>P19838</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     UniprotID_A UniprotID_B GeneName_A GeneName_B database\n",
       "61        P25025      P25025      CXCR2      CXCR2     APID\n",
       "644       Q12884      Q12884        FAP        FAP     APID\n",
       "814       P01100      P01100        FOS        FOS     APID\n",
       "855       P01100      P19838        FOS      NFKB1     APID\n",
       "1229      P19838      P19838      NFKB1      NFKB1     APID"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(29, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>UniprotID_A</th>\n",
       "      <th>UniprotID_B</th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>61</td>\n",
       "      <td>P25025</td>\n",
       "      <td>P25025</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>644</td>\n",
       "      <td>Q12884</td>\n",
       "      <td>Q12884</td>\n",
       "      <td>FAP</td>\n",
       "      <td>FAP</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>814</td>\n",
       "      <td>P01100</td>\n",
       "      <td>P01100</td>\n",
       "      <td>FOS</td>\n",
       "      <td>FOS</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>855</td>\n",
       "      <td>P01100</td>\n",
       "      <td>P19838</td>\n",
       "      <td>FOS</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1229</td>\n",
       "      <td>P19838</td>\n",
       "      <td>P19838</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   index UniprotID_A UniprotID_B GeneName_A GeneName_B database\n",
       "0     61      P25025      P25025      CXCR2      CXCR2     APID\n",
       "1    644      Q12884      Q12884        FAP        FAP     APID\n",
       "2    814      P01100      P01100        FOS        FOS     APID\n",
       "3    855      P01100      P19838        FOS      NFKB1     APID\n",
       "4   1229      P19838      P19838      NFKB1      NFKB1     APID"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seed_genes_interactome = pd.concat([b,cc])\n",
    "seed_genes_interactome.reset_index(inplace=True)\n",
    "print(seed_genes_interactome.shape)\n",
    "seed_genes_interactome.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniprotID_A</th>\n",
       "      <th>UniprotID_B</th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q13555</td>\n",
       "      <td>O60266</td>\n",
       "      <td>CAMK2G</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q8TAV4</td>\n",
       "      <td>O60266</td>\n",
       "      <td>STOML3</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P07550</td>\n",
       "      <td>O60266</td>\n",
       "      <td>ADRB2</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q9HBU9</td>\n",
       "      <td>O60266</td>\n",
       "      <td>POPDC2</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P17931</td>\n",
       "      <td>O60266</td>\n",
       "      <td>LGALS3</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  UniprotID_A UniprotID_B GeneName_A GeneName_B database\n",
       "0      Q13555      O60266     CAMK2G      ADCY3     APID\n",
       "1      Q8TAV4      O60266     STOML3      ADCY3     APID\n",
       "2      P07550      O60266      ADRB2      ADCY3     APID\n",
       "3      Q9HBU9      O60266     POPDC2      ADCY3     APID\n",
       "4      P17931      O60266     LGALS3      ADCY3     APID"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c=df_APID[(df_APID[\"GeneName_A\"].isin(Autoinflammatory_diseases)==False)] \n",
    "d = c[(c[\"GeneName_B\"].isin(Autoinflammatory_diseases)==True)]\n",
    "e=df_APID[(df_APID[\"GeneName_A\"].isin(Autoinflammatory_diseases)==True)] \n",
    "f = e[(e[\"GeneName_B\"].isin(Autoinflammatory_diseases)==False)]\n",
    "\n",
    "union_interactome_APID = pd.concat([d,f])\n",
    "union_interactome_APID.reset_index(inplace=True)\n",
    "union_interactome_APID.drop(\"index\", inplace = True, axis = 1)\n",
    "union_interactome_APID[\"database\"] = [\"APID\"]*union_interactome_APID.shape[0]\n",
    "union_interactome_APID.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#node1</th>\n",
       "      <th>node2</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PRKACB</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PRKACA</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRKACG</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PDE4C</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PDE10A</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #node1  node2 database\n",
       "0  PRKACB  ADCY3   STRING\n",
       "1  PRKACA  ADCY3   STRING\n",
       "2  PRKACG  ADCY3   STRING\n",
       "3   PDE4C  ADCY3   STRING\n",
       "4  PDE10A  ADCY3   STRING"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c=df_STRING[(df_STRING[\"#node1\"].isin(Autoinflammatory_diseases)==False)] \n",
    "d = c[(c[\"node2\"].isin(Autoinflammatory_diseases)==True)]\n",
    "e=df_STRING[(df_STRING[\"#node1\"].isin(Autoinflammatory_diseases)==True)] \n",
    "f = e[(e[\"node2\"].isin(Autoinflammatory_diseases)==False)]\n",
    "union_interactome_STRING = pd.concat([d,f])\n",
    "union_interactome_STRING.reset_index(inplace=True)\n",
    "union_interactome_STRING.drop(\"index\", inplace = True, axis = 1)\n",
    "union_interactome_STRING[\"database\"] = [\"STRING\"]*union_interactome_STRING.shape[0]\n",
    "union_interactome_STRING.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PRKACB</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PRKACA</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRKACG</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PDE4C</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PDE10A</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PDE4D</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PDE4B</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PDE7A</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PDE4A</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PDE6B</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PRKACB</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PRKACA</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PRKACG</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PDE4C</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PDE10A</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PDE4D</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PDE4B</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PDE7A</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PDE4A</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PDE6B</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>CXCL1</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>CXCL2</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ARRB1</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GNA14</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>FPR2</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PPP2CA</td>\n",
       "      <td>CXCR2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>H2AFX</td>\n",
       "      <td>DNMT3B</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>RBBP4</td>\n",
       "      <td>DNMT3B</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>RBBP7</td>\n",
       "      <td>DNMT3B</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>HDAC2</td>\n",
       "      <td>DNMT3B</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>IL2RA</td>\n",
       "      <td>NFATC1</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>IL2RA</td>\n",
       "      <td>GZMB</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>NFKB1</td>\n",
       "      <td>NFKBIA</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>NFKB1</td>\n",
       "      <td>BCL3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>NFKB1</td>\n",
       "      <td>RELB</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>NXPE1</td>\n",
       "      <td>MLH1</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412</th>\n",
       "      <td>OR5B21</td>\n",
       "      <td>GNAL</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>413</th>\n",
       "      <td>OR5B21</td>\n",
       "      <td>GNGT1</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>OR5B21</td>\n",
       "      <td>ADRBK2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>OR5B21</td>\n",
       "      <td>PRKACA</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>OR5B21</td>\n",
       "      <td>ADRBK1</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>OSM</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>TYK2</td>\n",
       "      <td>IL6</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>TYK2</td>\n",
       "      <td>OSM</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>OSMR</td>\n",
       "      <td>IL6</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>TYK2</td>\n",
       "      <td>STAT3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>STAT3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>STAT5B</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>PTPN2</td>\n",
       "      <td>STAT6</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>RAVER1</td>\n",
       "      <td>EIF4H</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>RAVER1</td>\n",
       "      <td>DDX1</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>427</th>\n",
       "      <td>RPS6KB1</td>\n",
       "      <td>MAPK1</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>SH2B3</td>\n",
       "      <td>KIT</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>SH2B3</td>\n",
       "      <td>SOCS3</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>SH2B3</td>\n",
       "      <td>EPOR</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>SH2B3</td>\n",
       "      <td>EPO</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>SH2B3</td>\n",
       "      <td>KITLG</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>SH2B3</td>\n",
       "      <td>GRB2</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>SH2B3</td>\n",
       "      <td>SH2B1</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>TNFRSF6B</td>\n",
       "      <td>FASLG</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>436 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    GeneName_A GeneName_B database\n",
       "0       PRKACB      ADCY3   STRING\n",
       "1       PRKACA      ADCY3   STRING\n",
       "2       PRKACG      ADCY3   STRING\n",
       "3        PDE4C      ADCY3   STRING\n",
       "4       PDE10A      ADCY3   STRING\n",
       "5        PDE4D      ADCY3   STRING\n",
       "6        PDE4B      ADCY3   STRING\n",
       "7        PDE7A      ADCY3   STRING\n",
       "8        PDE4A      ADCY3   STRING\n",
       "9        PDE6B      ADCY3   STRING\n",
       "10      PRKACB      ADCY3   STRING\n",
       "11      PRKACA      ADCY3   STRING\n",
       "12      PRKACG      ADCY3   STRING\n",
       "13       PDE4C      ADCY3   STRING\n",
       "14      PDE10A      ADCY3   STRING\n",
       "15       PDE4D      ADCY3   STRING\n",
       "16       PDE4B      ADCY3   STRING\n",
       "17       PDE7A      ADCY3   STRING\n",
       "18       PDE4A      ADCY3   STRING\n",
       "19       PDE6B      ADCY3   STRING\n",
       "20       CXCL1      CXCR2   STRING\n",
       "21       CXCL2      CXCR2   STRING\n",
       "22       ARRB1      CXCR2   STRING\n",
       "23       GNA14      CXCR2   STRING\n",
       "24        FPR2      CXCR2   STRING\n",
       "25      PPP2CA      CXCR2   STRING\n",
       "26       H2AFX     DNMT3B   STRING\n",
       "27       RBBP4     DNMT3B   STRING\n",
       "28       RBBP7     DNMT3B   STRING\n",
       "29       HDAC2     DNMT3B   STRING\n",
       "..         ...        ...      ...\n",
       "406      IL2RA     NFATC1   STRING\n",
       "407      IL2RA       GZMB   STRING\n",
       "408      NFKB1     NFKBIA   STRING\n",
       "409      NFKB1       BCL3   STRING\n",
       "410      NFKB1       RELB   STRING\n",
       "411      NXPE1       MLH1   STRING\n",
       "412     OR5B21       GNAL   STRING\n",
       "413     OR5B21      GNGT1   STRING\n",
       "414     OR5B21     ADRBK2   STRING\n",
       "415     OR5B21     PRKACA   STRING\n",
       "416     OR5B21     ADRBK1   STRING\n",
       "417       OSMR        OSM   STRING\n",
       "418       TYK2        IL6   STRING\n",
       "419       TYK2        OSM   STRING\n",
       "420       OSMR        IL6   STRING\n",
       "421       TYK2      STAT3   STRING\n",
       "422      PTPN2      STAT3   STRING\n",
       "423      PTPN2     STAT5B   STRING\n",
       "424      PTPN2      STAT6   STRING\n",
       "425     RAVER1      EIF4H   STRING\n",
       "426     RAVER1       DDX1   STRING\n",
       "427    RPS6KB1      MAPK1   STRING\n",
       "428      SH2B3        KIT   STRING\n",
       "429      SH2B3      SOCS3   STRING\n",
       "430      SH2B3       EPOR   STRING\n",
       "431      SH2B3        EPO   STRING\n",
       "432      SH2B3      KITLG   STRING\n",
       "433      SH2B3       GRB2   STRING\n",
       "434      SH2B3      SH2B1   STRING\n",
       "435   TNFRSF6B      FASLG   STRING\n",
       "\n",
       "[436 rows x 3 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "union_interactome_STRING.rename(columns={'#node1': 'GeneName_A', 'node2': 'GeneName_B'}, inplace=True)\n",
    "union_interactome_STRING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "u_list = []\n",
    "for i in union_interactome_STRING[\"GeneName_A\"].values:\n",
    "    if i in df_r[\"official gene symbol\"].values:\n",
    "        u_list.append(df_r[df_r[\"official gene symbol\"]==i][\"Uniprot AC\"].values[0])\n",
    "    else:\n",
    "        u_list.append(\"NaN\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "u1_list = []\n",
    "for i in union_interactome_STRING[\"GeneName_B\"].values:\n",
    "    if i in df_r[\"official gene symbol\"].values:\n",
    "        u1_list.append(df_r[df_r[\"official gene symbol\"]==i][\"Uniprot AC\"].values[0])\n",
    "    else:\n",
    "        u1_list.append(\"NaN\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "union_interactome_STRING[\"UniprotID_A\"] = u_list\n",
    "union_interactome_STRING[\"UniprotID_B\"] = u_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "columns_to_change = [\"UniprotID_A\", \"UniprotID_B\", \"GeneName_A\", \"GeneName_B\",\"database\"]\n",
    "union_interactome_STRING = union_interactome_STRING[columns_to_change]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "union_interactome = pd.concat([union_interactome_APID,union_interactome_STRING])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniprotID_A</th>\n",
       "      <th>UniprotID_B</th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "      <th>database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q13555</td>\n",
       "      <td>O60266</td>\n",
       "      <td>CAMK2G</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q8TAV4</td>\n",
       "      <td>O60266</td>\n",
       "      <td>STOML3</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P07550</td>\n",
       "      <td>O60266</td>\n",
       "      <td>ADRB2</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q9HBU9</td>\n",
       "      <td>O60266</td>\n",
       "      <td>POPDC2</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P17931</td>\n",
       "      <td>O60266</td>\n",
       "      <td>LGALS3</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>APID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  UniprotID_A UniprotID_B GeneName_A GeneName_B database\n",
       "0      Q13555      O60266     CAMK2G      ADCY3     APID\n",
       "1      Q8TAV4      O60266     STOML3      ADCY3     APID\n",
       "2      P07550      O60266      ADRB2      ADCY3     APID\n",
       "3      Q9HBU9      O60266     POPDC2      ADCY3     APID\n",
       "4      P17931      O60266     LGALS3      ADCY3     APID"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "union_interactome.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "union_interactome_APID_gene = union_interactome_APID[[\"GeneName_A\", \"GeneName_B\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "union_interactome_STRING_gene = union_interactome_STRING[[\"GeneName_A\", \"GeneName_B\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([['CAMK2G', 'ADCY3'],\n",
       "       ['STOML3', 'ADCY3'],\n",
       "       ['ADRB2', 'ADCY3'],\n",
       "       ..., \n",
       "       ['TYK2', 'ALYREF'],\n",
       "       ['TYK2', 'SIAH2'],\n",
       "       ['TYK2', 'UBE2L6']], dtype=object)"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a=union_interactome_APID_gene.values\n",
    "b=union_interactome_STRING_gene.values\n",
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for i in range(len(union_interactome_APID_gene)):\n",
    "#     for j in range(len(union_interactome_STRING_gene)):\n",
    "#         print(union_interactome_APID_gene[i, union_interactome_APID_gene.GeneName_A])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "intersection_interactome = union_interactome_APID.merge(union_interactome_STRING_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['UniprotID_A', 'UniprotID_B', 'GeneName_A', 'GeneName_B', 'database'], dtype='object')"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intersection_interactome.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniprotID_A</th>\n",
       "      <th>UniprotID_B</th>\n",
       "      <th>GeneName_A</th>\n",
       "      <th>GeneName_B</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P09341</td>\n",
       "      <td>P25025</td>\n",
       "      <td>CXCL1</td>\n",
       "      <td>CXCR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P09341</td>\n",
       "      <td>P25025</td>\n",
       "      <td>CXCL1</td>\n",
       "      <td>CXCR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q7Z6Z7</td>\n",
       "      <td>Q12884</td>\n",
       "      <td>HUWE1</td>\n",
       "      <td>FAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q7Z6Z7</td>\n",
       "      <td>Q12884</td>\n",
       "      <td>HUWE1</td>\n",
       "      <td>FAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Q04206</td>\n",
       "      <td>P19838</td>\n",
       "      <td>RELA</td>\n",
       "      <td>NFKB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Q04206</td>\n",
       "      <td>P19838</td>\n",
       "      <td>RELA</td>\n",
       "      <td>NFKB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Q04206</td>\n",
       "      <td>P19838</td>\n",
       "      <td>RELA</td>\n",
       "      <td>NFKB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Q04206</td>\n",
       "      <td>P19838</td>\n",
       "      <td>RELA</td>\n",
       "      <td>NFKB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Q14164</td>\n",
       "      <td>Q9BYX4</td>\n",
       "      <td>IKBKE</td>\n",
       "      <td>IFIH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Q14164</td>\n",
       "      <td>Q9BYX4</td>\n",
       "      <td>IKBKE</td>\n",
       "      <td>IFIH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Q15366</td>\n",
       "      <td>Q9BYX4</td>\n",
       "      <td>PCBP2</td>\n",
       "      <td>IFIH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Q15366</td>\n",
       "      <td>Q9BYX4</td>\n",
       "      <td>PCBP2</td>\n",
       "      <td>IFIH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Q7Z434</td>\n",
       "      <td>Q9BYX4</td>\n",
       "      <td>MAVS</td>\n",
       "      <td>IFIH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Q7Z434</td>\n",
       "      <td>Q9BYX4</td>\n",
       "      <td>MAVS</td>\n",
       "      <td>IFIH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Q9H1Y0</td>\n",
       "      <td>Q9BYX4</td>\n",
       "      <td>ATG5</td>\n",
       "      <td>IFIH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Q9H1Y0</td>\n",
       "      <td>Q9BYX4</td>\n",
       "      <td>ATG5</td>\n",
       "      <td>IFIH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Q9H1Y0</td>\n",
       "      <td>Q9BYX4</td>\n",
       "      <td>ATG5</td>\n",
       "      <td>IFIH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Q9H1Y0</td>\n",
       "      <td>Q9BYX4</td>\n",
       "      <td>ATG5</td>\n",
       "      <td>IFIH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>P42701</td>\n",
       "      <td>Q5VWK5</td>\n",
       "      <td>IL12RB1</td>\n",
       "      <td>IL23R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>P42701</td>\n",
       "      <td>Q5VWK5</td>\n",
       "      <td>IL12RB1</td>\n",
       "      <td>IL23R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Q9Y297</td>\n",
       "      <td>Q14145</td>\n",
       "      <td>BTRC</td>\n",
       "      <td>KEAP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Q9Y297</td>\n",
       "      <td>Q14145</td>\n",
       "      <td>BTRC</td>\n",
       "      <td>KEAP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>D6R9G5</td>\n",
       "      <td>Q14145</td>\n",
       "      <td>KLHL2</td>\n",
       "      <td>KEAP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>D6R9G5</td>\n",
       "      <td>Q14145</td>\n",
       "      <td>KLHL2</td>\n",
       "      <td>KEAP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Q04864</td>\n",
       "      <td>P19838</td>\n",
       "      <td>REL</td>\n",
       "      <td>NFKB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Q04864</td>\n",
       "      <td>P19838</td>\n",
       "      <td>REL</td>\n",
       "      <td>NFKB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>P33527</td>\n",
       "      <td>Q8IY67</td>\n",
       "      <td>ABCC1</td>\n",
       "      <td>RAVER1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>P33527</td>\n",
       "      <td>Q8IY67</td>\n",
       "      <td>ABCC1</td>\n",
       "      <td>RAVER1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>P54819</td>\n",
       "      <td>Q8IY67</td>\n",
       "      <td>AK2</td>\n",
       "      <td>RAVER1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>P54819</td>\n",
       "      <td>Q8IY67</td>\n",
       "      <td>AK2</td>\n",
       "      <td>RAVER1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>Q8NEV9</td>\n",
       "      <td>Q14213</td>\n",
       "      <td>IL27</td>\n",
       "      <td>EBI3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>Q8NEV9</td>\n",
       "      <td>Q14213</td>\n",
       "      <td>IL27</td>\n",
       "      <td>EBI3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>Q8NEV9</td>\n",
       "      <td>Q6UWB1</td>\n",
       "      <td>IL27</td>\n",
       "      <td>IL27RA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>Q8NEV9</td>\n",
       "      <td>Q6UWB1</td>\n",
       "      <td>IL27</td>\n",
       "      <td>IL27RA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>P01589</td>\n",
       "      <td>P14784</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>IL2RB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>P01589</td>\n",
       "      <td>P14784</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>IL2RB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>P01589</td>\n",
       "      <td>P40763</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>STAT3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>P01589</td>\n",
       "      <td>P40763</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>STAT3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>P01589</td>\n",
       "      <td>P40763</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>STAT3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>P01589</td>\n",
       "      <td>P40763</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>STAT3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>P01589</td>\n",
       "      <td>P60568</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>IL2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>P01589</td>\n",
       "      <td>P60568</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>IL2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>P19838</td>\n",
       "      <td>P25963</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>NFKBIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>P19838</td>\n",
       "      <td>P25963</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>NFKBIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>Q99650</td>\n",
       "      <td>P13725</td>\n",
       "      <td>OSMR</td>\n",
       "      <td>OSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>Q99650</td>\n",
       "      <td>P13725</td>\n",
       "      <td>OSMR</td>\n",
       "      <td>OSM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>P17706</td>\n",
       "      <td>P40763</td>\n",
       "      <td>PTPN2</td>\n",
       "      <td>STAT3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>P17706</td>\n",
       "      <td>P40763</td>\n",
       "      <td>PTPN2</td>\n",
       "      <td>STAT3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>P17706</td>\n",
       "      <td>P42226</td>\n",
       "      <td>PTPN2</td>\n",
       "      <td>STAT6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>P17706</td>\n",
       "      <td>P42226</td>\n",
       "      <td>PTPN2</td>\n",
       "      <td>STAT6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>P17706</td>\n",
       "      <td>P51692</td>\n",
       "      <td>PTPN2</td>\n",
       "      <td>STAT5B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>P17706</td>\n",
       "      <td>P51692</td>\n",
       "      <td>PTPN2</td>\n",
       "      <td>STAT5B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>P23443</td>\n",
       "      <td>P28482</td>\n",
       "      <td>RPS6KB1</td>\n",
       "      <td>MAPK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>P23443</td>\n",
       "      <td>P28482</td>\n",
       "      <td>RPS6KB1</td>\n",
       "      <td>MAPK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>Q9UQQ2</td>\n",
       "      <td>P10721</td>\n",
       "      <td>SH2B3</td>\n",
       "      <td>KIT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>Q9UQQ2</td>\n",
       "      <td>P10721</td>\n",
       "      <td>SH2B3</td>\n",
       "      <td>KIT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>Q9UQQ2</td>\n",
       "      <td>P62993</td>\n",
       "      <td>SH2B3</td>\n",
       "      <td>GRB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>Q9UQQ2</td>\n",
       "      <td>P62993</td>\n",
       "      <td>SH2B3</td>\n",
       "      <td>GRB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>O95407</td>\n",
       "      <td>P48023</td>\n",
       "      <td>TNFRSF6B</td>\n",
       "      <td>FASLG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>O95407</td>\n",
       "      <td>P48023</td>\n",
       "      <td>TNFRSF6B</td>\n",
       "      <td>FASLG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    UniprotID_A UniprotID_B GeneName_A GeneName_B\n",
       "0        P09341      P25025      CXCL1      CXCR2\n",
       "1        P09341      P25025      CXCL1      CXCR2\n",
       "2        Q7Z6Z7      Q12884      HUWE1        FAP\n",
       "3        Q7Z6Z7      Q12884      HUWE1        FAP\n",
       "4        Q04206      P19838       RELA      NFKB1\n",
       "5        Q04206      P19838       RELA      NFKB1\n",
       "6        Q04206      P19838       RELA      NFKB1\n",
       "7        Q04206      P19838       RELA      NFKB1\n",
       "8        Q14164      Q9BYX4      IKBKE      IFIH1\n",
       "9        Q14164      Q9BYX4      IKBKE      IFIH1\n",
       "10       Q15366      Q9BYX4      PCBP2      IFIH1\n",
       "11       Q15366      Q9BYX4      PCBP2      IFIH1\n",
       "12       Q7Z434      Q9BYX4       MAVS      IFIH1\n",
       "13       Q7Z434      Q9BYX4       MAVS      IFIH1\n",
       "14       Q9H1Y0      Q9BYX4       ATG5      IFIH1\n",
       "15       Q9H1Y0      Q9BYX4       ATG5      IFIH1\n",
       "16       Q9H1Y0      Q9BYX4       ATG5      IFIH1\n",
       "17       Q9H1Y0      Q9BYX4       ATG5      IFIH1\n",
       "18       P42701      Q5VWK5    IL12RB1      IL23R\n",
       "19       P42701      Q5VWK5    IL12RB1      IL23R\n",
       "20       Q9Y297      Q14145       BTRC      KEAP1\n",
       "21       Q9Y297      Q14145       BTRC      KEAP1\n",
       "22       D6R9G5      Q14145      KLHL2      KEAP1\n",
       "23       D6R9G5      Q14145      KLHL2      KEAP1\n",
       "24       Q04864      P19838        REL      NFKB1\n",
       "25       Q04864      P19838        REL      NFKB1\n",
       "26       P33527      Q8IY67      ABCC1     RAVER1\n",
       "27       P33527      Q8IY67      ABCC1     RAVER1\n",
       "28       P54819      Q8IY67        AK2     RAVER1\n",
       "29       P54819      Q8IY67        AK2     RAVER1\n",
       "..          ...         ...        ...        ...\n",
       "98       Q8NEV9      Q14213       IL27       EBI3\n",
       "99       Q8NEV9      Q14213       IL27       EBI3\n",
       "100      Q8NEV9      Q6UWB1       IL27     IL27RA\n",
       "101      Q8NEV9      Q6UWB1       IL27     IL27RA\n",
       "102      P01589      P14784      IL2RA      IL2RB\n",
       "103      P01589      P14784      IL2RA      IL2RB\n",
       "104      P01589      P40763      IL2RA      STAT3\n",
       "105      P01589      P40763      IL2RA      STAT3\n",
       "106      P01589      P40763      IL2RA      STAT3\n",
       "107      P01589      P40763      IL2RA      STAT3\n",
       "108      P01589      P60568      IL2RA        IL2\n",
       "109      P01589      P60568      IL2RA        IL2\n",
       "110      P19838      P25963      NFKB1     NFKBIA\n",
       "111      P19838      P25963      NFKB1     NFKBIA\n",
       "112      Q99650      P13725       OSMR        OSM\n",
       "113      Q99650      P13725       OSMR        OSM\n",
       "114      P17706      P40763      PTPN2      STAT3\n",
       "115      P17706      P40763      PTPN2      STAT3\n",
       "116      P17706      P42226      PTPN2      STAT6\n",
       "117      P17706      P42226      PTPN2      STAT6\n",
       "118      P17706      P51692      PTPN2     STAT5B\n",
       "119      P17706      P51692      PTPN2     STAT5B\n",
       "120      P23443      P28482    RPS6KB1      MAPK1\n",
       "121      P23443      P28482    RPS6KB1      MAPK1\n",
       "122      Q9UQQ2      P10721      SH2B3        KIT\n",
       "123      Q9UQQ2      P10721      SH2B3        KIT\n",
       "124      Q9UQQ2      P62993      SH2B3       GRB2\n",
       "125      Q9UQQ2      P62993      SH2B3       GRB2\n",
       "126      O95407      P48023   TNFRSF6B      FASLG\n",
       "127      O95407      P48023   TNFRSF6B      FASLG\n",
       "\n",
       "[128 rows x 4 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intersection_interactome.drop(\"database\",axis=1,inplace=True)\n",
    "intersection_interactome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
